### ISSUE TITLE
Sometimes poll with None timeout returns with no events

### ISSUE THREAD

**Issue by dtacalau** [tokio-rs/mio#1205]
I noticed this behavior when writing tests, so I wrote a simple test to isolate it.

This is a bug because according to poll doc:
```
A `timeout` of `None` means that
    /// `poll` will block until a readiness event has been received.
```

**Comment by dtacalau** [comment]
Repro test:

```
fn repro() {
    // Test to verify issue 1205 does not reproduce again
    init();

    let mut poll = Poll::new().unwrap();
    let mut events = Events::with_capacity(128);

    let mut listener = TcpListener::bind(any_local_address()).unwrap();

    poll.registry()
        .register(&mut listener, ID1, Interest::READABLE)
        .unwrap();

    poll.poll(&mut events, Some(std::time::Duration::from_millis(0)))
        .unwrap();
    assert!(events.iter().count() == 0);

    let _stream = TcpStream::connect(listener.local_addr().unwrap()).unwrap();

    poll.registry().deregister(&mut listener).unwrap();


    poll.poll(&mut events, None).unwrap();
    assert!(events.iter().count() > 1);
}
```

**Comment by dtacalau** [comment]
Fixed by #1206
